### ISSUE TITLE
Feature request: don't include signature by default.

### ISSUE THREAD

**Issue by robstewart57** [djcb/mu#327]
mu4e is working beautifully for me. There is one thing I am unsure how to achieve. I am able to tell mu4e to read my signature like:

```
(setq message-signature-file "~/.emacs.d/.signature")
```

However, most of my emails are informal and would prefer not to have the signature in place by default. When composing an email, I very much like the `C-c C-w` combination for inserting my signature.

Is there a way for me to disable signature inclusion by default, and only add it when I need it?

**Comment by Kabouik** [comment]
What would be the best way to do that now that `mu4e-compose-signature-auto-include` has been deprecated? It seems using `message-signature` is now the recommended way to set a signature for `mu4e`, but it will be auto-inserted all the time.

I would like to pre-format the signature using `message-signature`, without auto-inserting it in all new compose buffers. Those two things are different and were supported before, but I don't see how to do that now.

**Comment by djcb** [comment]
We're bound by what `message` does, which includes a signature unconditionally if set. So to avoid auto-inserting, you need to ensure that `message-signature` is `nil` when you create a message, and set it afterwards.

`message-signature`  can be a function, so you could try making it a function that returns your signature when in `mu4e-compose-mode` and returns nil otherwise.

**Comment by Kabouik** [comment]
I see, thanks, I'll give that a try!

**Comment by Kabouik** [comment]
> `message-signature` can be a function, so you could try making it a function that returns your signature when in `mu4e-compose-mode` and returns nil otherwise.

That was a great idea but I realized later that it is not possible to top post and have the signature above the quoted message. I understand that this is a standard and everything below `--` is to be considered signature, but unfortunately there are situations whereby following the standard won't fly if collaborators don't know or follow it. Consequently, I decided not to use `--` so as to not break email citations/history, hoping that this way I at least won't misuse the real signature system, and just sign below my messages (above citations) with more informal short signatures autocompleted with `tempo`.
